tests/test_zuora_interrupted_sync.py

```diff
@@ -0,0 +1,163 @@
+        second_sync_records = runner.get_records_from_target_output()
+        second_sync_bookmarks = menagerie.get_state(conn_id)
+        LOGGER.info("second_sync_record_count = %s \n second_sync_bookmarks = %s", second_sync_record_count, second_sync_bookmarks)
+
+        # Run sync after interruption
+        final_state = menagerie.get_state(conn_id)
+        currently_syncing = final_state.get('current_stream')
+
+        # Checking resuming the sync resulted in a successfully saved state
+        with self.subTest():
+
+            # Verify sync is not interrupted by checking currently_syncing in the state for sync
+            self.assertIsNone(currently_syncing)
+
+            # Verify bookmarks are saved
+            self.assertIsNotNone(final_state.get('bookmarks'))
+
+        # Stream level assertions
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+                # Expected values
+                expected_replication_method = self.expected_replication_method()[stream]
+                replication_key = next(iter(self.expected_replication_keys()[stream]))
+
+                # Collect information for assertions from syncs 1 & 2 base on expected values
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+                second_sync_messages = [record.get('data') for record in
+                                        second_sync_records.get(
+                                            stream, {}).get('messages', [])
+                                        if record.get('action') == 'upsert']
+                first_bookmark_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                LOGGER.info("first_bookmark_value =%s \n second_bookmark_value =%s", first_bookmark_value, second_bookmark_value)
+
+                # Final bookmark after interrupted sync
+                final_stream_bookmark = final_state['bookmarks'][stream]
+
+                if expected_replication_method == self.INCREMENTAL:
+                    interrupted_bookmark_value = interrupted_sync_states['bookmarks'][stream]
+                    if stream in completed_streams:
+                        # Verify at least 1 record was replicated in the second sync
+                        self.assertGreaterEqual(second_sync_count,
+                                            1, 
+                                            msg="Incorrect bookmarking for {0}, at least one or more record should be replicated".format(stream))
+
+                    elif stream == interrupted_sync_states.get('current_stream', None):
+                        # For interrupted stream records sync count should be less equals
+                        self.assertLessEqual(second_sync_count,
```

Comment:
Just a suggestion, here we can count the number of remaining records and have assertion like `self.assertGreaterEqual(second_sync_count, remaining_records)`.